packages/ragbits-document-search/src/ragbits/document_search/ingestion/processor_strategies/batched.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+from collections.abc import Sequence
+
+from ragbits.document_search.documents.document import Document, DocumentMeta
+from ragbits.document_search.documents.element import Element
+from ragbits.document_search.documents.sources import Source
+from ragbits.document_search.ingestion.document_processor import DocumentProcessorRouter
+from ragbits.document_search.ingestion.providers.base import BaseProvider
+
+from .base import ProcessingExecutionStrategy
+
+
+class BatchedAsyncProcessing(ProcessingExecutionStrategy):
```

Comment:
question: have you measured the performance difference between batched and sequential strategies ?